twitch.py

```diff
@@ -177,16 +177,30 @@ class TwitchHLSStreamWriter(HLSStreamWriter):
     reader: "TwitchHLSStreamReader"
     stream: "TwitchHLSStream"
 
+    def __init__(self, *args, **kwargs) -> None:
+        self._reexec_counter = 0
+        super().__init__(*args, **kwargs)
+
     def should_filter_sequence(self, sequence: TwitchSequence):  # type: ignore[override]
         if self.stream.reexec_on_ad and sequence.segment.ad:
             log.info("Encountered an ad segment, re-execing to retrieve a new playlist")
+            self._reexec_counter += 1
```

Comment:
Looks good, but after a second look won't this be reset to 0 after exec? We would need to store the counter outside of the process or read it from an argument passed to the process.

Reply:
Yes. Good catch. Passing the counter as an argument/file/environment variable to the next process seems messy. Since streamlink offers adding a timestamp as one of its meta variables, I think that should be the de facto solution.

Perhaps a note in the README saying that a timestamp should be added to the output file (using streamlink's meta variables) when using `--twitch-reexec-on-ad` would be a good idea.

Reply:
Okay. I will add that to the readme and credit you in the commit description. Apologies for not catching this in the issue originally.